### ISSUE TITLE
This implementation violates the invariant that the VRGDA should never reach 0

### ISSUE THREAD

**Issue by 0xAlcibiades** [transmissions11/VRGDAs#18]
![image](https://user-images.githubusercontent.com/89996683/201547685-393016f1-5695-41fe-af34-7d9b5f5be8d0.png)

Noticed this during invariant testing of an Art Gobblers adjacent project. As we approach infinity, this implementation rounds down. The conjecture is that it should round up to 1 wei.

Credit to @mevbandit @MarcusWentz @horsefacts for the help in discovering this.

![image](https://user-images.githubusercontent.com/89996683/201547814-e6f0e1c9-f6e7-4054-be0b-7ada85f284ce.png)

**Comment by transmissions11** [comment]
i dont think not reaching 0 is necessarily an invariant, whats the use case you have in mind where it needs to round up?

**Comment by 0xAlcibiades** [comment]
Looking at the equation, the function should never reach zero. I noticed it because we hit divide by zeros during invariant test and that function rounding up was one of the invariants.

**Comment by transmissions11** [comment]
we have finite precision, beyond a certain point both rounding up and rounding down are "wrong" behavior mathematically

**Comment by 0xAlcibiades** [comment]
Agree, but my conjecture is it's *always* a non-zero value, and so rounding up is more correct. Agree there is no right choice here though :).

**Comment by transmissions11** [comment]
i am a little biased cuz its a bit scary changing things to differ from the gobbler implementation, but i do kinda agree with you

specifically if you, say, wanted to sell something only to people who can get access to some token (say the token is normally soulbound or something) if you round down you allow people without access to that token to purchase

**Comment by transmissions11** [comment]
also iirc this led to some weird behavior in lower elos of monaco

**Comment by transmissions11** [comment]
@0xAlcibiades if you want to make a PR for this i think i'd be willing to merge it thru 👍

**Comment by 0xAlcibiades** [comment]
Alright, let me take a look at that mañana :)

**Comment by transmissions11** [comment]
awesome <3
